src/api/operator/numpy/np_matrix_op.cc

```diff
@@ -490,7 +490,10 @@ MXNET_REGISTER_API("_npi.diag")
   const nnvm::Op* op = Op::Get("_npi_diag");
   nnvm::NodeAttrs attrs;
   op::NumpyDiagParam param;
-  param.k = args[1].operator int();
+  if (features::is_enabled(features::INT64_TENSOR_SIZE))
+    param.k = args[1].operator int64_t();
+  else
+    param.k = args[1].operator int();
```

Comment:
why not just use `index_t`? if enabled it will be int64 else int32

Reply:
there is no index_t operator in include/mxnet/runtime/packed_func.h so we need to branch like this

Reply:
change here: https://github.com/apache/incubator-mxnet/blob/master/src/operator/numpy/np_matrix_op-inl.h#L1253

Reply:
it's already changed